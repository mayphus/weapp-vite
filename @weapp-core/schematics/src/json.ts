import { z } from 'zod'
import { zodToJsonSchema } from 'zod-to-json-schema'

// https://developers.weixin.qq.com/miniprogram/dev/reference/configuration/app.html

export const sharedSchema = z.object({
  usingComponents: z.object({}).optional(),
  componentFramework: z.string().optional(),
})

export const pageAndComponentSharedSchema = z.object({
  componentPlaceholder: z.object({}).optional(),
}).merge(sharedSchema)

export const appAndPageSharedSchema = z.object({
  style: z.string().optional(),
  singlePage: z.object({}).optional(),
  enablePassiveEvent: z.object({}).or(z.boolean()).optional(),
  renderer: z.enum(['webview', 'skyline']).optional().default('webview'),
  rendererOptions: z.object({
    skyline: z.object({
      defaultDisplayBlock: z.boolean().optional().default(false),
      defaultContentBox: z.boolean().optional().default(false),
      disableABTest: z.boolean().optional().default(false),
    }).optional(),
  }).optional(),
}).merge(sharedSchema)

export const windowSchema = z.object({
  navigationBarBackgroundColor: z.string().optional().default('#000000'),
  navigationBarTextStyle: z.string().optional().default('white'),
  navigationBarTitleText: z.string().optional(),
  navigationStyle: z.enum(['default', 'custom']).optional().default('default'), // z.string()
  homeButton: z.boolean().optional().default(false),
  backgroundColor: z.string().optional().default('#ffffff'),
  backgroundTextStyle: z.enum(['dark', 'light']).optional().default('dark'),
  backgroundColorTop: z.string().optional().default('#ffffff'),
  backgroundColorBottom: z.string().optional().default('#ffffff'),
  enablePullDownRefresh: z.boolean().optional().default(false),
  onReachBottomDistance: z.number().optional().default(50),
  pageOrientation: z.enum(['portrait', 'auto', 'landscape']).optional().default('portrait'),
  restartStrategy: z.enum(['homePage', 'homePageAndLatestPage']).optional().default('homePage'),
  initialRenderingCache: z.enum(['static', 'dynamic']).optional(),
  visualEffectInBackground: z.enum(['none', 'hidden']).optional().default('none'),
  handleWebviewPreload: z.enum(['static', 'manual', 'auto']).optional().default('static'),
})

export const tabBarSchema = z.object({
  color: z.string(),
  selectedColor: z.string(),
  backgroundColor: z.string(),
  borderStyle: z.enum(['black', 'white']).optional().default('black'),
  list: z.array(
    z.object({
      pagePath: z.string(),
      text: z.string(),
      iconPath: z.string().optional(),
      selectedIconPath: z.string().optional(),
    }),
  ).min(2).max(5),
  position: z.enum(['bottom', 'top']).optional().default('bottom'),
  custom: z.boolean().optional(),
})

export const networkTimeoutSchema = z.object({
  request: z.number().optional().default(60000),
  connectSocket: z.number().optional().default(60000),
  uploadFile: z.number().optional().default(60000),
  downloadFile: z.number().optional().default(60000),
})

export const subpackageSchema = z.object({
  root: z.string().optional(),
  name: z.string().optional(),
  pages: z.string().array().optional(),
  independent: z.boolean().optional(),
  entry: z.string().optional(),
})

export const AppSchema = z
  .object({
    // https://developers.weixin.qq.com/miniprogram/dev/reference/configuration/app.html#entryPagePath
    entryPagePath: z.string().optional().describe('指定小程序的默认启动路径（首页），常见情景是从微信聊天列表页下拉启动、小程序列表启动等。如果不填，将默认为 pages 列表的第一项。不支持带页面路径参数。'),
    pages: z.array(z.string()).describe('用于指定小程序由哪些页面组成，每一项都对应一个页面的 路径（含文件名） 信息。文件名不需要写文件后缀，框架会自动去寻找对应位置的 .json, .js, .wxml, .wxss 四个文件进行处理。'),
    window: windowSchema.optional(),
    tabBar: tabBarSchema.optional(),
    networkTimeout: networkTimeoutSchema.optional(),
    debug: z.boolean().optional(),
    functionalPages: z.boolean().optional(),
    subpackages: z.array(subpackageSchema).optional(),
    workers: z.string().optional(),
    requiredBackgroundModes: z.array(z.string()).optional(),
    requiredPrivateInfos: z.array(z.string()).optional(),
    plugins: z.object({}).optional(),
    preloadRule: z.object({}).optional(),
    resizable: z.boolean().optional(),
    permission: z.object({}).optional(),
    sitemapLocation: z.string().optional(),
    useExtendedLib: z.object({}).optional(),
    entranceDeclare: z.object({}).optional(),
    darkmode: z.boolean().optional(),
    themeLocation: z.string().optional(),
    lazyCodeLoading: z.string().optional(),
    supportedMaterials: z.object({}).optional(),
    serviceProviderTicket: z.string().optional(),
    embeddedAppIdList: z.array(z.string()).optional(),
    halfPage: z.object({}).optional(),
    debugOptions: z.object({}).optional(),
    resolveAlias: z.object({}).optional(),
    miniApp: z.object({}).optional(),
    static: z.object({}).optional(),
    convertRpxToVw: z.boolean().optional(),
  })
  .merge(appAndPageSharedSchema)
  .describe('全局配置, 小程序根目录下的 app.json 文件用来对微信小程序进行全局配置。')

export const PageSchema = z
  .object({
    backgroundColorContent: z.string().optional().default('#RRGGBBAA'),
    disableScroll: z.boolean().optional().default(false),
    styleIsolation: z.enum(['page-isolated', 'page-apply-shared', 'page-shared']).optional(),
  })
  .merge(appAndPageSharedSchema)
  .merge(pageAndComponentSharedSchema)
  .merge(windowSchema)
  .describe('页面配置, 支持对单个页面进行配置，可以在页面对应的 .json 文件来对本页面的表现进行配置')

export const ComponentSchema = z
  .object({
    component: z.boolean().default(true),
    styleIsolation: z.enum(['isolated', 'apply-shared', 'shared']).optional().default('isolated'),
    componentGenerics: z.object({}).optional(),
  })
  .merge(pageAndComponentSharedSchema)
  .describe('自定义组件配置')

export const AppJsonSchema = zodToJsonSchema(AppSchema)

export const PageJsonSchema = zodToJsonSchema(PageSchema)

export const ComponentJsonSchema = zodToJsonSchema(ComponentSchema)
