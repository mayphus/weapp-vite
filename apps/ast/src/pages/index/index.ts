import { x } from '../../utils'
// @ts-ignore
// eslint-disable-next-line ts/no-require-imports
const y = require('../../utils/async')

Page(
  {
    onReady() {
      console.log(x, y)
    },
  },
)
