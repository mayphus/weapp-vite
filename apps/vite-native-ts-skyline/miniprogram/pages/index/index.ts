import Toast from 'tdesign-miniprogram/toast/index'

Component({
  data: {
    motto: 'Hello World',
    userInfo: {
      nickName: '',
    },
    hasUserInfo: false,
    canIUseGetUserProfile: wx.canIUse('getUserProfile'),
    canIUseNicknameComp: wx.canIUse('input.type.nickname'),
  },
  methods: {
    showSuccessToast() {
      Toast({
        context: this,
        selector: '#t-toast',
        message: '成功文案',
        theme: 'success',
        direction: 'column',
      })
    },
  },
})
