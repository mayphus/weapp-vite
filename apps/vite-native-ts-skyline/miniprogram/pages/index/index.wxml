<!-- index.wxml -->
<navigation-bar title="Weixin" back="{{false}}" color="black" background="#FFF"></navigation-bar>
<view class="text-[200rpx] text-[#4343ff] ">
 11212123111111
</view>