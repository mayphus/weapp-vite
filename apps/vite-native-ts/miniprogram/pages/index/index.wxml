<HelloWorld></HelloWorld>
<page-meta>
  <navigation-bar title="新标题" loading="true" front-color="#ffffff" background-color="#000000" color-animation-duration="2000" color-animation-timing-func="easeIn" />
</page-meta>
<try-custom-tab-bar-2/>