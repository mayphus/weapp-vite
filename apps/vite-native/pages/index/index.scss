.s {
  .a {
    color: turquoise;
  }
}

.user-motto {
  font-size: 92rpx;
}
