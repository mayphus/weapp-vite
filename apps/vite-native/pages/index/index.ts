// index.js
// import { formatTime } from '../../utils/util'
import logoUrl from '@/assets/logo.png'
// const { formatTime } = require('../../utils/util')
import { formatTime } from '../../utils/util'

console.log('-------------', import.meta.env)
const { MP_PLATFORM } = import.meta.env
console.log(MP_PLATFORM)
const defaultAvatarUrl = 'https://mmbiz.qpic.cn/mmbiz/icTdbqWNOwNRna42FI242Lcia07jQodd2FJGIYQfG0LAJGFxM4FbnQP6yfMxBgJ0F3YRqJCJ1aPAK2dQagdusBZg/0'
Page({
  data: {
    logoUrl,
    motto: JSON.stringify(import.meta.env, null, 2),
    message: 'Hello MINA!',
    userInfo: {
      avatarUrl: defaultAvatarUrl,
      nickName: '',
    },
    array: [1, 2, 3, 4, 5],
    hasUserInfo: false,
    canIUseGetUserProfile: wx.canIUse('getUserProfile'),
    canIUseNicknameComp: wx.canIUse('input.type.nickname'),
    time: formatTime(new Date()),
    staffA: { firstName: 'Hulk', lastName: 'Hu' },
    staffB: { firstName: 'Shang', lastName: 'You' },
    staffC: { firstName: 'Gideon', lastName: 'Lin' },
    flag: true,
  },
  bindViewTap() {
    wx.navigateTo({
      url: '../logs/logs',
    })
  },
  switchTemplate(e) {
    console.log(this.data.flag)
    this.setData({
      flag: !this.data.flag,
    })
  },
  onChooseAvatar(e) {
    const { avatarUrl } = e.detail
    const { nickName } = this.data.userInfo
    this.setData({
      'userInfo.avatarUrl': avatarUrl,
      'hasUserInfo': nickName && avatarUrl && avatarUrl !== defaultAvatarUrl,
    })
  },
  onInputChange(e) {
    const nickName = e.detail.value
    const { avatarUrl } = this.data.userInfo
    this.setData({
      'userInfo.nickName': nickName,
      'hasUserInfo': nickName && avatarUrl && avatarUrl !== defaultAvatarUrl,
    })
  },
  getUserProfile() {
    // 推荐使用wx.getUserProfile获取用户信息，开发者每次通过该接口获取用户个人信息均需用户确认，开发者妥善保管用户快速填写的头像昵称，避免重复弹窗
    wx.getUserProfile({
      desc: '展示用户信息', // 声明获取用户个人信息后的用途，后续会展示在弹窗中，请谨慎填写
      success: (res) => {
        console.log(res)
        this.setData({
          userInfo: res.userInfo,
          hasUserInfo: true,
        })
      },
    })
  },
  hello(...args: any[]) {
    console.log(...args)
  },
})
