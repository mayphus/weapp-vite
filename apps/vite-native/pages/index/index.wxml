<!-- index.wxml -->
<template name="staffName">
  <view>FirstName: {{firstName}}, LastName: {{lastName}}</view>
</template>
<navigation-bar></navigation-bar>
<image src="../../logo.png" class="w-40 h-40"></image>
<ice-avatar></ice-avatar>
<scroll-view scroll-y type="list">
  <view class="p-2 border m-2 rounded">{{ motto }}</view>
  <view class="bg-[#c50d69] text-[10px] text-[#123456] w-[323px] {{true?'h-[30px]':'h-[45px]'}}">
    bg-[#7d7ac2] text-[10px] text-[#123456] w-[323px]
  </view>
  <view class="p-2 border">{{message}}</view>
  <view wx:for="{{array}}">{{item}}</view>
  <template is="staffName" data="{{...staffA}}"></template>
  <template is="staffName" data="{{...staffB}}"></template>
  <template is="staffName" data="{{...staffC}}"></template>
  <template bindtap="switchTemplate" is="{{flag? 'odd': 'staffName'}}" data="{{...staffC}}"></template>
  <t-divider></t-divider>
  <view class="user-motto text-pink-200 bg-slate-100">{{m1.message}}</view>
  <wxs module="wxs" src="./test.wxs"></wxs>
  <view id="tapTest" data-hi="Weixin" bindtap="{{wxs.tapName}}">Click me!</view>
</scroll-view>
<template name="odd">
  <view>odd</view>
</template>
<template name="even">
  <view>even</view>
</template>
<view class="button-example">
  <t-button bind:tap="switchTemplate" theme="primary" size="large">填充按钮</t-button>
  <t-button bind:click="switchTemplate" theme="light" size="large">填充按钮</t-button>
  <t-button size="large">填充按钮</t-button>
</view>
<block wx:for="{{[1, 2, 3, 4, 5]}}">
  <template is="{{item % 2 == 0 ? 'even' : 'odd'}}" />
</block>
<wxs module="m1">
var msg = "hello world";

module.exports.message = msg;
</wxs>