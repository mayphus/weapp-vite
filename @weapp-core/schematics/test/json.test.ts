import fs from 'fs-extra'
import path from 'pathe'
import { generateJson } from '@/index'
import { AppJsonSchema, ComponentJsonSchema, PageJsonSchema, SitemapJsonSchema, ThemeJsonSchema } from '../scripts/json'

describe('json', () => {
  describe('jsonSchema ', () => {
    it('appJsonSchema', () => {
      expect(AppJsonSchema).toMatchSnapshot()
    })

    it('pageJsonSchema', () => {
      expect(PageJsonSchema).toMatchSnapshot()
    })

    it('componentJsonSchema', () => {
      expect(ComponentJsonSchema).toMatchSnapshot()
    })

    it('themeJsonSchema', () => {
      expect(ThemeJsonSchema).toMatchSnapshot()
    })

    it('sitemapJsonSchema', () => {
      expect(SitemapJsonSchema).toMatchSnapshot()
    })
  })

  describe('generateJson', () => {
    it('generateJson app', async () => {
      const json = generateJson('app')
      expect(json).matchSnapshot()
      json && await fs.writeFile(path.resolve(__dirname, './fixtures/json/app.json'), json, 'utf8')
    })

    it('generateJson app js', async () => {
      const json = generateJson('app', 'js')
      expect(json).matchSnapshot()
      json && await fs.writeFile(path.resolve(__dirname, './fixtures/json/app.json.js'), json, 'utf8')
    })

    it('generateJson app ts', async () => {
      const json = generateJson('app', 'ts')
      expect(json).matchSnapshot()
      json && await fs.writeFile(path.resolve(__dirname, './fixtures/json/app.json.ts'), json, 'utf8')
    })

    it('generateJson component', async () => {
      const json = generateJson('component')
      expect(json).matchSnapshot()
      json && await fs.writeFile(path.resolve(__dirname, './fixtures/json/component.json'), json, 'utf8')
    })

    it('generateJson component js', async () => {
      const json = generateJson('component', 'js')
      expect(json).matchSnapshot()
      json && await fs.writeFile(path.resolve(__dirname, './fixtures/json/component.json.js'), json, 'utf8')
    })

    it('generateJson component ts', async () => {
      const json = generateJson('component', 'ts')
      expect(json).matchSnapshot()
      json && await fs.writeFile(path.resolve(__dirname, './fixtures/json/component.json.ts'), json, 'utf8')
    })

    it('generateJson page', async () => {
      const json = generateJson('page')
      expect(json).matchSnapshot()
      json && await fs.writeFile(path.resolve(__dirname, './fixtures/json/page.json'), json, 'utf8')
    })

    it('generateJson page js', async () => {
      const json = generateJson('page', 'js')
      expect(json).matchSnapshot()
      json && await fs.writeFile(path.resolve(__dirname, './fixtures/json/page.json.js'), json, 'utf8')
    })

    it('generateJson page ts', async () => {
      const json = generateJson('page', 'ts')
      expect(json).matchSnapshot()
      json && await fs.writeFile(path.resolve(__dirname, './fixtures/json/page.json.ts'), json, 'utf8')
    })
  })
})
