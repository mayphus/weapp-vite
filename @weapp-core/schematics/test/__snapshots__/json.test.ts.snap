// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`json > jsonSchema  > appJsonSchema 1`] = `
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "additionalProperties": {},
  "description": "全局配置, 小程序根目录下的 app.json 文件用来对微信小程序进行全局配置。",
  "properties": {
    "$schema": {
      "type": "string",
    },
    "componentFramework": {
      "type": "string",
    },
    "convertRpxToVw": {
      "type": "boolean",
    },
    "darkmode": {
      "type": "boolean",
    },
    "debug": {
      "type": "boolean",
    },
    "debugOptions": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "embeddedAppIdList": {
      "items": {
        "type": "string",
      },
      "type": "array",
    },
    "enablePassiveEvent": {
      "anyOf": [
        {
          "additionalProperties": {},
          "properties": {},
          "type": "object",
        },
        {
          "type": "boolean",
        },
      ],
    },
    "entranceDeclare": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "entryPagePath": {
      "description": "指定小程序的默认启动路径（首页），常见情景是从微信聊天列表页下拉启动、小程序列表启动等。如果不填，将默认为 pages 列表的第一项。不支持带页面路径参数。",
      "type": "string",
    },
    "functionalPages": {
      "type": "boolean",
    },
    "halfPage": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "lazyCodeLoading": {
      "type": "string",
    },
    "miniApp": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "networkTimeout": {
      "additionalProperties": {},
      "properties": {
        "connectSocket": {
          "default": 60000,
          "type": "number",
        },
        "downloadFile": {
          "default": 60000,
          "type": "number",
        },
        "request": {
          "default": 60000,
          "type": "number",
        },
        "uploadFile": {
          "default": 60000,
          "type": "number",
        },
      },
      "type": "object",
    },
    "pages": {
      "description": "用于指定小程序由哪些页面组成，每一项都对应一个页面的 路径（含文件名） 信息。文件名不需要写文件后缀，框架会自动去寻找对应位置的 .json, .js, .wxml, .wxss 四个文件进行处理。",
      "items": {
        "type": "string",
      },
      "type": "array",
    },
    "permission": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "plugins": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "preloadRule": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "renderer": {
      "default": "webview",
      "enum": [
        "webview",
        "skyline",
      ],
      "type": "string",
    },
    "rendererOptions": {
      "additionalProperties": {},
      "properties": {
        "skyline": {
          "additionalProperties": {},
          "properties": {
            "defaultContentBox": {
              "default": false,
              "type": "boolean",
            },
            "defaultDisplayBlock": {
              "default": false,
              "type": "boolean",
            },
            "disableABTest": {
              "default": false,
              "type": "boolean",
            },
          },
          "type": "object",
        },
      },
      "type": "object",
    },
    "requiredBackgroundModes": {
      "items": {
        "type": "string",
      },
      "type": "array",
    },
    "requiredPrivateInfos": {
      "items": {
        "type": "string",
      },
      "type": "array",
    },
    "resizable": {
      "type": "boolean",
    },
    "resolveAlias": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "serviceProviderTicket": {
      "type": "string",
    },
    "singlePage": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "sitemapLocation": {
      "type": "string",
    },
    "static": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "style": {
      "type": "string",
    },
    "subpackages": {
      "items": {
        "additionalProperties": {},
        "properties": {
          "entry": {
            "type": "string",
          },
          "independent": {
            "type": "boolean",
          },
          "name": {
            "type": "string",
          },
          "pages": {
            "items": {
              "type": "string",
            },
            "type": "array",
          },
          "root": {
            "type": "string",
          },
        },
        "type": "object",
      },
      "type": "array",
    },
    "supportedMaterials": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "tabBar": {
      "additionalProperties": {},
      "properties": {
        "backgroundColor": {
          "type": "string",
        },
        "borderStyle": {
          "default": "black",
          "enum": [
            "black",
            "white",
          ],
          "type": "string",
        },
        "color": {
          "type": "string",
        },
        "custom": {
          "type": "boolean",
        },
        "list": {
          "items": {
            "additionalProperties": {},
            "properties": {
              "iconPath": {
                "type": "string",
              },
              "pagePath": {
                "type": "string",
              },
              "selectedIconPath": {
                "type": "string",
              },
              "text": {
                "type": "string",
              },
            },
            "required": [
              "pagePath",
              "text",
            ],
            "type": "object",
          },
          "maxItems": 5,
          "minItems": 2,
          "type": "array",
        },
        "position": {
          "default": "bottom",
          "enum": [
            "bottom",
            "top",
          ],
          "type": "string",
        },
        "selectedColor": {
          "type": "string",
        },
      },
      "required": [
        "color",
        "selectedColor",
        "backgroundColor",
        "list",
      ],
      "type": "object",
    },
    "themeLocation": {
      "type": "string",
    },
    "useExtendedLib": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "usingComponents": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "window": {
      "additionalProperties": {},
      "properties": {
        "backgroundColor": {
          "default": "#ffffff",
          "type": "string",
        },
        "backgroundColorBottom": {
          "default": "#ffffff",
          "type": "string",
        },
        "backgroundColorTop": {
          "default": "#ffffff",
          "type": "string",
        },
        "backgroundTextStyle": {
          "default": "dark",
          "enum": [
            "dark",
            "light",
          ],
          "type": "string",
        },
        "enablePullDownRefresh": {
          "default": false,
          "type": "boolean",
        },
        "handleWebviewPreload": {
          "default": "static",
          "enum": [
            "static",
            "manual",
            "auto",
          ],
          "type": "string",
        },
        "homeButton": {
          "default": false,
          "type": "boolean",
        },
        "initialRenderingCache": {
          "enum": [
            "static",
            "dynamic",
          ],
          "type": "string",
        },
        "navigationBarBackgroundColor": {
          "default": "#000000",
          "type": "string",
        },
        "navigationBarTextStyle": {
          "default": "white",
          "type": "string",
        },
        "navigationBarTitleText": {
          "type": "string",
        },
        "navigationStyle": {
          "default": "default",
          "enum": [
            "default",
            "custom",
          ],
          "type": "string",
        },
        "onReachBottomDistance": {
          "default": 50,
          "type": "number",
        },
        "pageOrientation": {
          "default": "portrait",
          "enum": [
            "portrait",
            "auto",
            "landscape",
          ],
          "type": "string",
        },
        "restartStrategy": {
          "default": "homePage",
          "enum": [
            "homePage",
            "homePageAndLatestPage",
          ],
          "type": "string",
        },
        "visualEffectInBackground": {
          "default": "none",
          "enum": [
            "none",
            "hidden",
          ],
          "type": "string",
        },
      },
      "type": "object",
    },
    "workers": {
      "type": "string",
    },
  },
  "required": [
    "pages",
  ],
  "type": "object",
}
`;

exports[`json > jsonSchema  > componentJsonSchema 1`] = `
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "additionalProperties": {},
  "description": "自定义组件配置",
  "properties": {
    "$schema": {
      "type": "string",
    },
    "component": {
      "default": true,
      "type": "boolean",
    },
    "componentFramework": {
      "type": "string",
    },
    "componentGenerics": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "componentPlaceholder": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "styleIsolation": {
      "default": "isolated",
      "enum": [
        "isolated",
        "apply-shared",
        "shared",
      ],
      "type": "string",
    },
    "usingComponents": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
  },
  "type": "object",
}
`;

exports[`json > jsonSchema  > pageJsonSchema 1`] = `
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "additionalProperties": {},
  "description": "页面配置, 支持对单个页面进行配置，可以在页面对应的 .json 文件来对本页面的表现进行配置",
  "properties": {
    "$schema": {
      "type": "string",
    },
    "backgroundColor": {
      "default": "#ffffff",
      "type": "string",
    },
    "backgroundColorBottom": {
      "default": "#ffffff",
      "type": "string",
    },
    "backgroundColorContent": {
      "default": "#RRGGBBAA",
      "type": "string",
    },
    "backgroundColorTop": {
      "default": "#ffffff",
      "type": "string",
    },
    "backgroundTextStyle": {
      "default": "dark",
      "enum": [
        "dark",
        "light",
      ],
      "type": "string",
    },
    "componentFramework": {
      "type": "string",
    },
    "componentPlaceholder": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "disableScroll": {
      "default": false,
      "type": "boolean",
    },
    "enablePassiveEvent": {
      "anyOf": [
        {
          "additionalProperties": {},
          "properties": {},
          "type": "object",
        },
        {
          "type": "boolean",
        },
      ],
    },
    "enablePullDownRefresh": {
      "default": false,
      "type": "boolean",
    },
    "handleWebviewPreload": {
      "default": "static",
      "enum": [
        "static",
        "manual",
        "auto",
      ],
      "type": "string",
    },
    "homeButton": {
      "default": false,
      "type": "boolean",
    },
    "initialRenderingCache": {
      "enum": [
        "static",
        "dynamic",
      ],
      "type": "string",
    },
    "navigationBarBackgroundColor": {
      "default": "#000000",
      "type": "string",
    },
    "navigationBarTextStyle": {
      "default": "white",
      "type": "string",
    },
    "navigationBarTitleText": {
      "type": "string",
    },
    "navigationStyle": {
      "default": "default",
      "enum": [
        "default",
        "custom",
      ],
      "type": "string",
    },
    "onReachBottomDistance": {
      "default": 50,
      "type": "number",
    },
    "pageOrientation": {
      "default": "portrait",
      "enum": [
        "portrait",
        "auto",
        "landscape",
      ],
      "type": "string",
    },
    "renderer": {
      "default": "webview",
      "enum": [
        "webview",
        "skyline",
      ],
      "type": "string",
    },
    "rendererOptions": {
      "additionalProperties": {},
      "properties": {
        "skyline": {
          "additionalProperties": {},
          "properties": {
            "defaultContentBox": {
              "default": false,
              "type": "boolean",
            },
            "defaultDisplayBlock": {
              "default": false,
              "type": "boolean",
            },
            "disableABTest": {
              "default": false,
              "type": "boolean",
            },
          },
          "type": "object",
        },
      },
      "type": "object",
    },
    "restartStrategy": {
      "default": "homePage",
      "enum": [
        "homePage",
        "homePageAndLatestPage",
      ],
      "type": "string",
    },
    "singlePage": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "style": {
      "type": "string",
    },
    "styleIsolation": {
      "enum": [
        "page-isolated",
        "page-apply-shared",
        "page-shared",
      ],
      "type": "string",
    },
    "usingComponents": {
      "additionalProperties": {},
      "properties": {},
      "type": "object",
    },
    "visualEffectInBackground": {
      "default": "none",
      "enum": [
        "none",
        "hidden",
      ],
      "type": "string",
    },
  },
  "type": "object",
}
`;
